_posts/2024-01-12-kasten-openshift-monitoring.markdown

```diff
@@ -0,0 +1,815 @@
+        },
+        "value": [
+          1705357729.062,
+          "0"
+        ]
+      }
+    ]
+  }
+}
+```
+
+The non integer value `1.0526315789473684` is because even if we see a change from 2 to 4 on a time range this is seen as a gradual change on the 10 minutes. If instead of `[10]` you choose `[5m]` the value would be higher because the change is seen as quicker hence more brutal.
+
+```
+        "value": [
+          1705356529.062,
+          "1.25"
+        ]
+```
+
+You can also create a graph in the promethus UI to get a better visibility of this signal. 
+
+Within the 10 minutes of the backup the number of failed backup increase : 
+![Signal start](/images/posts/2024-01-12-kasten-openshift-monitoring/start-signal.png)
+
+But after 10 minutes the `increase` eval again to zero:
+![Signal has stopped](/images/posts/2024-01-12-kasten-openshift-monitoring/stop-signal.png)
+
+From an alerting point of view, what you need to do at the end is to capture when this expression will be true 
+
+```
+increase(catalog_actions_count{status='failed'}[10m])>0
+```
+
+This expression is a binary selector selector, if it evaluates to true it will return the instant vector that satifie the condition `> 0`.
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "mcourcy-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "mcourcy-pacman-backup",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "catalog-svc",
+          "status": "failed",
+          "type": "backup"
+        },
+        "value": [
+          1705356529.062,
+          "1.0526315789473684"
+        ]
+      }
+    ]
+  }
+}   
+```
+
+In other words the exact result we get before, otherwise it just return an empty set
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": []
+  }
+}
+```
+
+Getting an empty resultset means that the number of failed action did not increase in the last 10 minutes.
+
+## Garbage collection of the metrics 
+
+Even long after the backup failure if you execute the request 
+```
+"query=increase(catalog_actions_count{policy='mcourcy-pacman-backup',status='failed'}[10m])"
+```
+
+You'll notice the date are still recent and within the last 10 minutes despite the failure is now over. The catalog continue to publish the metrics or the actions, and prometheus continue to scrape them. 
+
+This is because backup actions has not been removed and will be removed after the restorepoint will be removed.
+
+# Capturing other interesting failure 
+
+You can use the Kasten grafana dashboard to obtain other interesting alert 
```

Comment:
```suggestion
It's worth noting that Kasten deploys an instance of Grafana when it is deployed. You can use the Kasten Grafana dashboard to obtain other interesting alerts
```

Reply:
"when it is deployed." is redundant!

Reply:
I like the suggestion but I agree it's redundant, I will find something